src/web.ts

```
@@ -0,0 +1,24 @@
+import { WebPlugin } from '@capacitor/core';
+import type { PrivacyScreenConfig, PrivacyScreenPlugin } from './definitions';
+
+export class PrivacyScreenWeb extends WebPlugin implements PrivacyScreenPlugin {
+  private enabled: boolean = false;
+
+  async enable(_config?: PrivacyScreenConfig): Promise<{ success: boolean }> {
+    this.enabled = true;
+    console.warn('Privacy Screen protection is not available on web platforms');
+    return { success: false };
+  }
+
+  async disable(): Promise<{ success: boolean }> {
+    this.enabled = false;
+    console.warn('Privacy Screen protection is not available on web platforms');
+    return { success: false };
+  }
```

Unsure if we should have return success false or instead throw an error if it's not available on web.

Also the `success` flag feels redundant to me; if `success=false`, should that instead be an error?

Unless this is to have an API that resembled what Identity Vault had; if so then it makes sense to me.

---

The `success` flag is somewhat common in Capacitor plugins, though you're right that it does seem redundant.

I think the rationale is that it's expected that since the native functionality for this plugin won't work on web, throwing errors would make it harder for developers to handle cross-platform code. The `success` flag makes it clear when an operation completed but didn't achieve its intended result. 

So I guess it fails gracefully and makes it so the developer doesn't have to write platform-specific error handling, but for actual errors (like invalid parameters or runtime failures), then we typically reject the calls / throw errors.

--

I think in this case we can go either way since they have their own pros / cons:
1. Remove the `success` flag and add more explicit errors with the caveat that developers will have to handle web errors more gracefully
2. Keep as-is and there are no explicit errors that they can catch when something fails. (We can add logging for debug purposes).

---

Thanks for the rationale explanation.

While I don't really like that the success=false won't have any reason on why it failed (particularly, Web won't receive an error message for plugin not being implemented) - I get what you mean with having an error that will always be returned and needed to be caught on web. So I'm okay with leaving as is.

Are the docs clear that this plugin is supported on native Android and iOS only?

---

Thanks for the rationale explanation.

While I don't really like that the success=false won't have any reason on why it failed (particularly, Web won't receive an error message for plugin not being implemented) - I get what you mean with having an error that will always be returned and needed to be caught on web. So I'm okay with leaving as is.

Are the docs clear that this plugin is supported on native Android and iOS only?

---

I will update the docs and make it clear 👍